python/ql/src/Security/CWE-327/InsecureDefaultProtocol.qhelp

```diff
@@ -34,20 +35,42 @@
           <p>
             Both of the cases above should be updated to use a secure protocol
             instead, for instance by specifying
-            <code>ssl_version=PROTOCOL_TLSv1_1</code> as a keyword argument.
+            <code>ssl_version=PROTOCOL_TLSv1_2</code> as a keyword argument.
+          </p>
+          <p>
+            The latter example can also be made secure by modifying the created
+            context before it is used to create a connection and so will not be
+            flagged by this query. However, if a connection is created before
+            the context has been secured (e.g. by setting the value of <code>minimum_version</code>),
+            then the code should be flagged by the query <code>py/insecure-protocol</code>.
           </p>
           <p>
             Note that <code>ssl.wrap_socket</code> has been deprecated in
-            Python 3.7. A preferred alternative is to use
-            <code>ssl.SSLContext</code>, which is supported in Python 2.7.9 and
-            3.2 and later versions.
+            Python 3.7. The recommended alternatives are:
           </p>
+          <ul>
+            <li><code>ssl.SSLContext</code> - supported in Python 2.7.9,
+                  3.2, and later versions</li>
+            <li><code>ssl.create_default_context</code> - a convenience function,
+                  supported in Python 3.4 and later versions.</li>
+          </ul>
+          <p>
+            Note also that, even using these alternatives, it is recommended to
+            ensure that a safe protocol is being used. The following code illustrates
+            how to use either flags (available since Python 3.2) or the `minimum_version`
```

Comment:
Suggest that this might be easier to read as:

```suggestion
            Even when you use these alternatives, you should
            ensure that a safe protocol is used. The following code illustrates
            how to use flags (available since Python 3.2) or the `minimum_version`
```